test/expr/expressions.swift

```diff
@@ -244,13 +244,12 @@ func test_lambda() {
   var a = { (value: Int) -> () in markUsed(value+1) }
 
   // A recursive lambda.
-  // FIXME: This should definitely be accepted.
-  var fib = { (n: Int) -> Int in
+  var fib: (Int) -> Int = { (n: Int) -> Int in
     if (n < 2) {
       return n
     }
     
-    return fib(n-1)+fib(n-2) // expected-error 2 {{variable used within its own initial value}}
+    return fib(n-1)+fib(n-2)
```

Comment:
This has to continue diagnosing; the variable can't be captured like this.

Reply:
Ah ok, I saw the fix-me above, so I updated the code. Without the explicit type annotation, it gets diagnosed. Perhaps I should look into the bodies of closures as well which have more than 1 expression.